internal/controller/genplan.go

```diff
@@ -118,14 +118,14 @@ func (r *TemporalWorkerDeploymentReconciler) generatePlan(
 	plan.DeleteDeployments = planResult.DeleteDeployments
 	plan.ScaleDeployments = planResult.ScaleDeployments
 
-	// Convert version config
-	if planResult.VersionConfig != nil {
-		plan.UpdateVersionConfig = &versionConfig{
-			conflictToken:  planResult.VersionConfig.ConflictToken,
-			versionID:      planResult.VersionConfig.VersionID,
-			setDefault:     planResult.VersionConfig.SetDefault,
-			rampPercentage: planResult.VersionConfig.RampPercentage,
-		}
+	// Convert version configs
+	for _, vcfg := range planResult.VersionConfigs {
```

Comment:
Is there a reason that `planResult.VersionConfigs` should be a different type than `plan.UpdateVersionConfigs`? If the fields are identical, we could just make them the same.

Although making them separate does allow us to add or remove fields from one or the other and not both in the future, if that is desired.

Anyways, this is fine for now, I'm just wondering about the reasoning for this verbose struct copy, and whether we can make it tighter!